townhall/myapi/views.py

```diff
@@ -7,14 +7,48 @@
 from rest_framework import status
 from rest_framework.response import Response
 from .services import VolunteerServices as volunteer_services
+from .services import OpportunityServices as opportunity_services
+from .serializers import OpportunitySerializer, VolunteerSerializer
 
 
 class VolunteerViewSet(viewsets.ModelViewSet):
 
+    @action(detail=False, methods=['get'], url_path='get_volunteer')
     def get_volunteer(self, request):
         volunteer_id = self.request.query_params.get('id')
 
         volunteer_obj = volunteer_services.get_volunteer(id=volunteer_id)
         if not volunteer_obj:
             return Response({"error": "Volunteer not found"}, status=status.HTTP_404_NOT_FOUND) 
-        return Response(dataclasses.asdict(volunteer_obj), status=status.HTTP_200_OK)
\ No newline at end of file
+        
+        serializer = VolunteerSerializer(volunteer_obj)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+class OpportunityViewSet(viewsets.ModelViewSet):
+    
+    @action(detail=False, methods=['get'], url_path='get_opportunity')
+    def get_opportunity(self, request):
+        opportunity_id = self.request.query_params.get('id')
+
+        opportunity_obj = opportunity_services.get_opportunity(id=opportunity_id)
+        if not opportunity_obj:
+            return Response({"error": "Opportunity not found"}, status=status.HTTP_404_NOT_FOUND)
+        
+        serializer = OpportunitySerializer(opportunity_obj)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    
+    @action(detail=False, methods=['get'], url_path='get_opportunity_all')
+    def get_opportunity_all(self,request):
+        opportunities = opportunity_services.get_opportunity_all()
+        if not opportunities:
+            return Response({"error": "Opportunities not found"}, status=status.HTTP_404_NOT_FOUND)
+        
+        serializer = OpportunitySerializer(opportunities, many=True)
+        return Response(serializer.data, status=status.HTTP_200_OK)
+    
+    @action(detail=False, methods=['delete'], url_path='delete_opportunity')
```

Comment:
I'll wait until the other PRs are merged before merging this change since this PR seems to include changes from the other PR

I will share this knowledge on Notion but as a whole when youre working on multiple tickets that have **dependency on one another** the cleanest way to make PR is to try to merge the PR to the other branch instead of all of them pointing at `main`. Here's an illustration to show 
<img width="403" alt="image" src="https://github.com/user-attachments/assets/fe79639c-272c-4ce6-a13c-2265c1e1b945">

Reply:
So in this case, this PR should point/merge to `tow-44-get-opportunity-all` instead of `main`